crates/proto/src/xfer/dns_handle.rs

```diff
@@ -62,16 +62,25 @@ pub trait DnsHandle: 'static + Clone + Send + Sync + Unpin {
     /// * `options` - options to use when constructing the message
     fn lookup(&self, query: Query, options: DnsRequestOptions) -> Self::Response {
         debug!("querying: {} {:?}", query.name(), query.query_type());
-        self.send(DnsRequest::new(build_message(query, options), options))
+        self.send(build_request(query, options))
```

Comment:
I'd prefer for this kind of thing to be in a different commit, since this one is large enough already. Also maybe `build_request()` should be a constructor on `DnsRequest` instead?

Reply:
I think I'd prefer to keep `build_request()` in this file for the time being, in order to keep the comment about randomizing the ID near the poll method it references.